python-package/lightgbm/compat.py

```diff
@@ -162,10 +105,3 @@ def _check_sample_weight(sample_weight, X, dtype=None):
     _LGBMAssertAllFinite = None
     _LGBMCheckClassificationTargets = None
     _LGBMComputeSampleWeight = None
-
-
-# DeprecationWarning is not shown by default, so let's create our own with higher level
-class LGBMDeprecationWarning(UserWarning):
```

Comment:
Could there be some user code downstream catching/filtering this one?

Reply:
Sorry, could you elaborate on what code you'd expect?

Reply:
It's probably ok, I don't think anything needs to be done :) What I meant is that you could say it's *technically* a breaking change since there can be user code like below:

```python
from lightgbm.compat import LGBBDeprecationWarning
import warnings
warnings.simplefilter('ignore', LGBMDeprecationWarning)
```

Reply:
Ah, OK, got it!

Yeah, after this PR is merged, one should change 
```
from lightgbm.compat import LGBMDeprecationWarning
```
to 

```
from lightgbm.basic import LGBMDeprecationWarning
```

But we are planing to include this PR in `3.2.0` (not `3.1.1` patch) version and it has `breaking` label in the changelog. So, I believe it is OK...